packages/ui/browser/src/ts/HtmlComponent.ts

```diff
@@ -74,122 +87,191 @@ export class HtmlComponent implements Destroyable {
+    }
 
-        this.parent = parent;
+    public static create(): HtmlComponent {
+        return new HtmlComponent();
+    }
+
+    protected set parent(parent: HtmlComponent | undefined) {
+        this.parent_ = parent;
         if (has(this.parent) && this.visibilityChangeListeners.size > 0) {
             this.parent.onVisibilityChange(this.myParentVisibilityListener);
         }
     }
 
+    protected get parent(): HtmlComponent | undefined {
+        return this.parent_;
+    }
+
+    /**
+     * Set the parent component of this component in order to propagate visibility changes bidirectionally.
+     */
+    public setParent(parent: HtmlComponent | undefined): this {
+        this.parent = parent;
+        return this;
+    }
+
+    protected set htmlContent(htmlContent: string | undefined) {
+        this.htmlContent_ = htmlContent;
+        this.maybeRenderAgain();
+    }
+
+    protected get htmlContent(): string | undefined {
+        return this.htmlContent_;
+    }
+
     /**
-     * @param container Either a query selector or a HTMLElement
-     * @param renderLocation The target render location relative to the container.
+     * See: https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML
+     */
+    public setHtmlContent(htmlContent: string): this {
+        this.htmlContent = htmlContent;
+        return this;
+    }
+
+    protected set classAttr(string: string | undefined) {
+        this.classAttr_ = string;
+        this.maybeRenderAgain();
+    }
+
+    protected get classAttr(): string | undefined {
+        return this.classAttr_;
+    }
+
+    /**
+     * See: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/class
+     */
+    public setClassAttribute(classAttr: string): this {
+        this.classAttr = classAttr;
+        return this;
+    }
+
+    protected set styleAttr(string: string | undefined) {
+        this.styleAttr_ = string;
+        this.maybeRenderAgain();
+    }
+
+    protected get styleAttr(): string | undefined {
+        return this.styleAttr_;
+    }
+
+    /**
+     * See: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/style
+     */
+    public setStyleAttribute(styleAttr: string): this {
+        this.styleAttr_ = styleAttr;
+        return this;
+    }
+
+    private maybeRenderAgain(): void {
+        this.isRendered() && this.destroy().then(() => this.renderTo(this.container!, this.renderLocation));
+    }
+
+    /**
+     * Renders to a {@link HtmlComponentContainer}.
      */
     public async renderTo(
-        container: string | HTMLElement | HtmlComponent,
+        container: HtmlComponentContainer,
         renderLocation = RenderLocation.IntoEnd
     ): Promise<this> {
+
+        const renderStart = Date.now();
+
+        if (!has(container)) {
+            throw new UndefinedContainerElementError();
+        } else {
+            this.container = container;
+            this.renderLocation = renderLocation;
+        }
```

Comment:
nit: No need for `else` block. :)
```suggestion
        }

        this.container = container;
        this.renderLocation = renderLocation;
```